src/cp/constraints/planed_shifts.py

```diff
@@ -0,0 +1,75 @@
+from . import Constraint
+from employee import Employee
+from shift import Shift
+from ..variables import Variable, EmployeeDayShiftVariable
+from ortools.sat.python.cp_model import CpModel
+import logging
+
+
+class PlannedShiftsConstraint(Constraint):
+    KEY = "planned-shifts"
+
+    SHIFT_MAPPING = {
+        "F": 0,
+        "Z": 1,
+        "S": 2,
+        "N": 3,
+        "Z60": 4,
+    }
```

Comment:
This should already be included in Shifts... but we need to watch out at this could change the forward rotation of shifts. I think a property on each shift like the `hidden` property on the employees makes sense.